Userland/Libraries/LibWebSocket/ConnectionInfo.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2021, The SerenityOS developers.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <AK/URL.h>
+#include <AK/Vector.h>
+#include <LibCore/Object.h>
+#include <LibCore/TCPSocket.h>
+#include <LibTLS/TLSv12.h>
+#include <LibWebSocket/Message.h>
+
+namespace WebSocket {
+
+class ConnectionInfo final {
+public:
+    ConnectionInfo(URL);
+
+    URL const& url() const { return m_url; }
+
+    String const& origin() const { return m_origin; }
+    void set_origin(String origin) { m_origin = origin; }
```

Comment:
Since we're already passing the string by value, would this (and the setters below) benefit from `move()`?

```suggestion
    void set_origin(String origin) { m_origin = move(origin); }
```

Reply:
Good point ! I'm still not that comfortable with move, but that's a place where it definitely should be :)
I also added them to the constructor (to move the URL), and passed the `ConnectionInfo` as a `const&` instead of a raw value in the `AbstractWebSocketImpl::connect()` and subclasses to prevent some additional copies.